class-two-factor-core.php

```diff
@@ -1902,30 +1901,20 @@ public static function user_two_factor_options( $user ) {
 	 * @return bool True if the provider was enabled, false otherwise.
 	 */
 	public static function enable_provider_for_user( $user_id, $new_provider ) {
-		$available_providers = self::get_providers();
```

Comment:
This method is very heavy as it will instantiate new set of provider classes on every call. That's why we introduced `get_provider_classes()` earlier which doesn't do any of that.